### PR TITLE
Feature/cross entropy loss

### PR DESCRIPTION
**Pull request by baykaner**
Implemented cross entropy loss function for Tensor.
Various math functions and SFINAE math/meta/type_traits edits were necessary.
ml/examples/basic_example now trains mnist using cross_entropy_loss

### CODE REVIEW

**Review by pierre-wilmot-fetch-ai — COMMENTED** [review]
Unit tests Please 🙏

**Review by pierre-wilmot-fetch-ai — DISMISSED** [review]

**Review by pierre-wilmot-fetch-ai — APPROVED** [review]
